File: cube-backend/app/schemas/log.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class OperationLogItem(BaseModel):
    id: int
    user_id: Optional[int] = None
    device_id: Optional[str] = None
    action: str
    detail: Optional[str] = None
    ip_address: Optional[str] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class OperationLogQuery(BaseModel):
    device_id: Optional[str] = None
    action: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)


class VoiceLogItem(BaseModel):
    id: int
    user_id: Optional[int] = None
    device_id: Optional[str] = None
    command_text: Optional[str] = None
    intent: Optional[str] = None
    executed: bool = False
    response_text: Optional[str] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class VoiceLogQuery(BaseModel):
    device_id: Optional[str] = None
    intent: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)


class VoiceLogCreate(BaseModel):
    device_id: str
    command_text: str
    intent: Optional[str] = None
    executed: bool = False
    response_text: Optional[str] = None

File: cube-tianmu/src/api/log.ts
import api from './index'

export interface OperationLogItem {
  id: number
  user_id: number | null
  device_id: string | null
  action: string
  detail: string | null
  ip_address: string | null
  created_at: string
}

export interface VoiceLogItem {
  id: number
  user_id: number | null
  device_id: string | null
  command_text: string
  intent: string | null
  executed: boolean
  response_text: string | null
  created_at: string
}

export interface PageResult<T> {
  items: T[]
  total: number
  page: number
  page_size: number
}

export const getOperationLogs = (params?: {
  device_id?: string
  action?: string
  page?: number
  page_size?: number
}) => {
  return api.get('/log/operation', { params }) as Promise<PageResult<OperationLogItem>>
}

export const getVoiceLogs = (params?: {
  device_id?: string
  intent?: string
  page?: number
  page_size?: number
}) => {
  return api.get('/log/voice', { params }) as Promise<PageResult<VoiceLogItem>>
}
